product_list_price_from_pricelist/models/product_pricelist.py

```diff
@@ -0,0 +1,142 @@
+from odoo import api, models
+from odoo.tools import config
+
+
+class ProductPricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _update_product_price_from_pricelist(self, pricelist_items=None):
+        self.ensure_one()
+        all_templates = self.env["product.template"]
+        if not self._can_update_product_price():
+            return False
+        pricelist_items = self.item_ids if not pricelist_items else pricelist_items
+        for item in pricelist_items:
+            templates = item._get_all_templates_from_pricelist_item()
+            all_templates |= templates
```

Comment:
Why putting an intermediate variable that is not used for anything more?

```suggestion
            all_templates |= item._get_all_templates_from_pricelist_item()
```